build/commands/scripts/checkSecurity.js

```diff
@@ -7,7 +7,7 @@ const execSync = require('child_process').execSync
 
 // Grep for any usage of innerHTML / document.write.
 // Ping @security-team before changing this.
-const cmd = "git grep -i --extended-regexp '(innerHTML|document.write)' ':(exclude)*test.cc' ':(exclude)test/*' ':(exclude)*.json' ':(exclude)build/*' ':(exclude)*browsertest*.cc'"
+const cmd = "git grep -i -E --not -e '^\\s*//' --and -e '(innerHTML|document.write)' ':(exclude)*test.cc' ':(exclude)test/*' ':(exclude)*.json' ':(exclude)build/*' ':(exclude)*browsertest*.cc'"
```

Comment:
I think there's a typo in the regex rule and should be `'^\s*//'` instead of `'^\\s*//'`, otherwise it will never ignore C++ / Javascript comments because `\\s` does not translate to "empty space", but to the `\` character followed by an `s`.

Other than that, I think the rule is simple enough and valid as long as we're happy with it knowing that the grep rule is now included some C++ specific exception, even though the `git grep` command applies to many other different types of files (yet I think that, in practice, it's probably good enough for this purpose).

Reply:
@mariospr i originally had it as `^\s*//` but it didn't work, i think because `\` is a special character in JavaScript. So therefore, we need to escape `\` itself for the sake of the JavaScript interpreter. See https://javascript.info/regexp-escaping for instance.

Reply:
You can make a test file like this, add it with `git add` and check that it works:

```
// innerHTML
/ innerHTML
innerHTML
```

the latter two cause failures, whereas the first doesn't
```

Reply:
Ah! Sorry, didn't realize this was a JS file, my fault. I was testing the git grep command from command line and then it obviously didn't work, but if you're escaping it because this is part of a JS script then it does make sense.

LGTM then